packages/clerk-js/src/core/clerk.ts

```diff
@@ -1482,7 +1485,10 @@ export default class Clerk implements ClerkInterface {
       { options: this.#options, displayConfig: this.#environment.displayConfig },
       false,
     );
-    return this.buildUrlWithAuth(appendAsQueryParams(signInOrUpUrl, opts));
+
+    return this.buildUrlWithAuth(
+      appendUrlsAsQueryParams(appendAsQueryParams(signInOrUpUrl, options?.initialValues || {}), opts),
```

Comment:
❓ These two functions (`appendUrlsAsQueryParams` and `appendAsQueryParams`) have similar names and are used together. This looks a little confusing. What's the difference between them?

Reply:
The `appendUrlsAsQueryParams` was named `appendAsQueryParams` but was only used for urls (doing more than just appending), so the naming was wrong. I renamed it to reflect its usage and added my own `appendAsQueryParams`.